email/parser.go

```diff
@@ -275,15 +282,24 @@ func extractSkylinks(input []byte) []string {
 	// range over the string line by line and extract potential skylinks
 	sc := bufio.NewScanner(bytes.NewBuffer(input))
 	for sc.Scan() {
-		line := strings.ReplaceAll(sc.Text(), " ", "")
-		for _, match := range skylinkRE.FindStringSubmatch(line) {
-			if validateSkylink32RE.Match([]byte(match)) {
-				maybeSkylinks = append(maybeSkylinks, match)
-				continue
-			}
-			if validateSkylink64RE.Match([]byte(match)) {
-				maybeSkylinks = append(maybeSkylinks, match)
-				continue
+		for _, line := range []string{
+			sc.Text(),
+			strings.ReplaceAll(sc.Text(), " ", ""),
```

Comment:
Is there a reason for running it once over the original line and once over the line without spaces?
If so, is it because there are skylinks with spaces in them? Also if there are I suppose we can't extract multiple of those in the same line?

I don't think that's an edge case that is worthwhile to pursue right now but I'm trying to understand this here.